testing/execution.ts

```diff
@@ -677,6 +679,24 @@ export async function executeSyncFormulaFromPackDefSingleIteration(
   ) as Promise<GenericSyncFormulaResult>;
 }
 
+/**
+ * Transforms a result from a sync formula execution to one that can be passed to executeGetPermissions.
+ *
+ * @hidden
+ */
+export function transformSyncFormulaResultToGetPermissionsRequest(
```

Comment:
Ah this is unfortunate. This is because when you call a sync formula in a test you get back normalized values?

Originally when we created the unittest framework, you got back pre-normalized values, which to me made the most sense because the pack maker is not supposed to care about normalization, it's a thing Coda does on their behalf but they don't really need to know or care about. At some point we made the unittest framework more robust or moved where normalization happens and as a side effect the return values started coming back normalized. So now developers do need to confront this detail.

It's probably worth seeing briefly whether there's a way to get back pre-normalized values during testing so that developers can just ignore this entirely. It might not be feasible though if things are tightly wound up in there.

Reply:
Yeah, correct - when you call `executeSyncFormulaFromPackDef` or `executeSyncFormulaFromPackDefSingleIteration` that returns the normalized values. (There are some examples in the `packs` repo.) This happens in `testing/execution.ts` in `findAndExecutePackFunction()`.

If we want to test permissions with the same values as passed in during ingestions then we need to untransform the result. However it looks like @alexd-codaio had a todo to revert these transforms:

https://github.com/coda/packs-sdk/blob/0d5ab265782ed6f9bad9b3fc56d838844be11db1/testing/execution.ts#L95-L96

However if we bypass transforms then extraneous properties aren't filtered out, and we also have to set `validateResult: false` since that seems broken too. So there's a bit of work, we might want to transform and untransform back implicitly (perhaps add a new flag).

Reply:
Oh good find. Ignoring extraneous properties for a second, could you get by without  transformSyncFormulaResultToGetPermissionsRequest() if you just used `useDeprecatedResultNormalization = false`?

That said, transforming and untransforming I think does simulate what happens in `coda`, so it seems reasonable to do that and not expose an untransformer. I'd rather we just put that behavior behind useDeprecatedResultNormalization, use it explicitly for now, and then announce that we're changing the default.

Reply:
Err sorry, I guess transform + untransform would become the behavior when useDeprecatedResultNormalization is _false_. You would explicitly set it to false in your new tests for now, and then we can make it part of this announcement and  then change the default.

Reply:
Sounds good, let me update. And yeah easier to work with than dealing with a new helper function, thanks for pointing me here to have another look at the code.

Reply:
Updated! Using this on https://github.com/coda/packs/pull/4494.